llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -943,11 +955,24 @@ bool LoopVectorizationLegality::canVectorizeInstrs() {
       if (CI && !VFDatabase::getMappings(*CI).empty())
         VecCallVariantsFound = true;
 
+      auto canWidenInstruction = [this](Instruction const &Inst) {
+        Type *InstTy = Inst.getType();
+        if (isa<CallInst>(Inst) && isa<StructType>(InstTy) &&
+            canWidenCallReturnType(InstTy)) {
+          StructVecVecCallFound = true;
```

Comment:
I don't want loop `LoopVectorizationLegality` to fail. The intention is `LoopVectorizationLegality` sees these calls are vectorizable (and returns normally). It's just the LoopVectorizer is incomplete and can't handle them yet. I think @david-arm has done something similar in the past. 

This flag is meant to be easily removed once it's supported.

Reply:
I guess the alternative to having a new flag would be to look through all instructions in LoopVectorize::processLoop and bail out if we encounter types that return a structure. It's less efficient though.

It looks like the main goal here is to teach LoopVectorizationLegality that this type of operation is legal, but emit a message to show the vectoriser doesn't have the capability yet. Just trying to split up bigger patches to aid the review process and also getting some limited defence of the tests.

Reply:
Right I see there is an already a TODO to remove the `StructVecVecCallFound`, so as long as it is temporary that's sounds OK to me. It would be good if we could make this clear with a comment here as well to avoid confusion for future readers of the code, as in most cases the checks in LVL are  directly tied to what we can generate code for.

Another alternative may worth considering is the future would be to bail out during VPlan construction if we encounter something that we cannot generate recipes for. There are a few cases where we already do that.

Reply:
Added a TODO to remove the flag here too (and yes the flag is immediately removed in #109833, which I rebased yesterday).